pandas/tools/merge.py

```diff
@@ -436,7 +427,7 @@ def _merger(x, y):
     if by is not None:
         result, groupby = _groupby_and_merge(by, on, left, right,
                                              lambda x, y: _merger(x, y),
-                                             check_duplicates=check_duplicates)
+                                             check_duplicates=False)
```

Comment:
Note that `check_duplicates` is always `False` now, since it is no longer needed.